language/move-stdlib/sources/vector.move

```diff
@@ -146,7 +146,26 @@ module std::vector {
         pragma intrinsic = true;
     }
 
-    /// Swap the `i`th element of the vector `v` with the last element and then pop the element.
+    /// Insert `e` at position `i` in the vector `v`, shifting the old `v[i]` and all subsequent elements to the right.
+    /// This is O(n) and preserves ordering of elements in the vector.
+    /// Aborts if `i` is out of bounds.
+    public fun insert<Element>(v: &mut vector<Element>, e: Element, i: u64) {
+        let len = length(v);
+        // i out of bounds; abort
+        if (i >= len) abort EINDEX_OUT_OF_BOUNDS;
+
+        push_back(v, e);
+        swap(v, len, i);
+        while (i < len) {
+            i = i + 1;
+            swap(v, i, len);
```

Comment:
This always does an unnecessary `swap(v, len, len)` at the end. Either decrease by one, or even cleaner:

```rust
push_back(v,e);
while (i < len) {
  swap(v, i, len);
  i = i + 1;
}
```

Reply:
Good catch!